src/sledo/optimiser.py

```diff
@@ -1,188 +1,129 @@
         if not self.data_dir.exists():
             self.data_dir.mkdir()
 
-        # Load input file, if one was passed.
-        if input_file:
-            self.load_input_file(path_to_input_file=input_file)
-
-    def load_input_file(self, path_to_input_file):
-        """Load a MOOSE input file to serve as a base for optimisation. Save a
-        copy to the data directory.
-        """
-        # Load input file.
-        filename_to_load = str(path_to_input_file)
-        moose_file = pyhit.load(filename_to_load)
-
-        # Save a copy.
-        filename_to_save = str(self.data_dir / "input_unmodified.i")
-        pyhit.write(filename_to_save, moose_file)
-
-    def generate_modified_file(self, filename, new_params):
-        """Generate a MOOSE input file by modifying the new_params specified in
-        new_params. Save the new file to the data directory.
+        # Instantiate Tuner object.
+        self.tuner = tune.Tuner(
+            self.evaluation_function,
+            tune_config=tune.TuneConfig(
+                mode="min",
+                metric=metric,
+                search_alg=self.search_alg,
+                num_samples=max_total_trials,
+            ),
+            run_config=train.RunConfig(
+                name=self.name,
+            ),
+            param_space=search_space,
+        )
 
-        Parameters
-        ----------
-        filename (str):
-            filename for the modified input file
-        new_params (list of dicts):
-            list of dictionaries containing information on which new_params to
-            modify
-        """
-        # Load unmodified input file from data directory.
-        path_to_base_input_file = self.data_dir / "input_unmodified.i"
-        if not path_to_base_input_file.is_file():
-            print("Base input file not found. Use load_input_file() first.")
-        root = pyhit.load(str(path_to_base_input_file))
-
-        # Locate and modify each parameter.
-        for param, value in zip(new_params.keys(), new_params.values()):
-            hit_block = self.hit_blocks[param]
-            hit_name = self.hit_names[param]
-            if hit_block:
-                block = moosetree.find(
-                    root, func=lambda n: n.fullpath == hit_block
-                )
-            else:
-                block = root
-            block[hit_name] = value
-            block.setComment(hit_name, "Modified by SLEDO")
-
-        # Write modified input file to data directory.
-        path_to_modified_file = self.data_dir / f"{filename}.i"
-        pyhit.write(path_to_modified_file, root)
-
-        return path_to_modified_file
-
-    def run_optimisation_loop(
-        self,
-        objective_name="",
-        force=False,
-        minimise=False,
-        parameter_constraints=None,
-        outcome_constraints=None,
-        max_iter=25,
-        min_acqf=None,
-    ):
+    def run_optimisation(self) -> ResultGrid:
         """Run the optimisation loop.
 
```

Comment:
Consider moving the stopping criterea to be an arguement to the run_optimisation (loop) method, else make it straightforward to modify stopping criterea and continue an optimisation run. There may be cases where the user wants to run, say, 50 iterations, then decides to run up to 200 more until hitting some threshold on the expected improvement. We should support this sort of flexibility on runs and maximise ease of reusing any data gathered so far in the aquisition function.